chromadb/api/segment.py

```diff
@@ -339,11 +344,18 @@ def _add(
         metadatas: Optional[Metadatas] = None,
         documents: Optional[Documents] = None,
         uris: Optional[URIs] = None,
-    ) -> bool:
+    ) -> AddResult:
         self._quota.static_check(metadatas, documents, embeddings, str(collection_id))
         coll = self._get_collection(collection_id)
         self._manager.hint_use_collection(collection_id, t.Operation.ADD)
 
+        ids = self.generate_ids_when_not_present(
+            ids=ids,
+            n_documents=len(documents) if documents is not None else 0,
```

Comment:
should just compute this length inside the generator function itself - this encapsulates handling of `None` and allows us to be defensive.